app/models/concerns/hostext/search.rb

```diff
@@ -109,11 +109,17 @@ def search_by_user(key, operator, value)
         end
         key_name = User.connection.quote_column_name(clean_key)
         condition = sanitize_sql_for_conditions(["#{key_name} #{operator} ?", value_to_sql(operator, value)])
-        users = User.where(condition)
-        hosts = users.map(&:hosts).flatten
-        opts  = hosts.empty? ? "< 0" : "IN (#{hosts.map(&:id).join(',')})"
 
-        {:conditions => " hosts.id #{opts} " }
+        user_ids = User.where(condition).reorder(nil).distinct.pluck(:id)
```

Comment:
If one does not join anything, which I do not see done in this context, the usage of distinct is redundant as the primary key is defined by being a **unique** identifier, therefore can only occur once.